File: Array/443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int out = 0;
        if(chars.length == 0){
            return out;
        }

        char last = chars[0];
        int count = 1;
        
        int index = 0;
        for(int i =1;i<chars.length;i++) {
            char curr = chars[i];
            
            if(last!=curr) {
                chars[index++] = last;
                if(count > 1) {
                    for(char c: (count+"").toCharArray()) {
                        chars[index++] = c;
                    }
                    count = 1;
                }
                last = curr;
                
            } else {
                count++;
            }
        }

        chars[index++] = last;
        if(count > 1) {
            for(char c: (count+"").toCharArray()) {
                chars[index++] = c;
            }
        }

        return index;


        
    }
}

File: Array/443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        read = 0
        write = 0

        while read < len(chars):
            start = read

            while read < len(chars) and chars[start] == chars[read]:
                read+=1
            count = read - start

            chars[write] = chars[start]
            write+=1
            if count > 1:
                s_count = str(count)
                chars[write: write+len(s_count)] = s_count
                write+= len(s_count)
        return write